ddtrace/_trace/processor/__init__.py

```diff
@@ -324,6 +324,14 @@ def on_span_finish(self, span):
 
                 if should_partial_flush:
                     log.debug("Partially flushing %d spans for trace %d", num_finished, span.trace_id)
+                    if not finished:
+                        log_msg = "Partial flush triggered but no spans to flush (was tracer reconfigured?)"
+                        # Avoid potential crashes if, for some reason, we have no spans to flush even though
+                        # should_partial_flush is True
+                        if config._telemetry_enabled:
```

Comment:
I don't think we need the enabled check here? 🤔

Reply:
the telemetry writer will check if it is enabled, and only queue the log message if it is enabled:

https://github.com/DataDog/dd-trace-py/blob/8c2758ad21cbbf24ab5cc3db351cb2b7cad0ff44/ddtrace/internal/telemetry/writer.py#L630

Reply:
Huh. The import of `telemetry` is conditional over at https://github.com/DataDog/dd-trace-py/pull/9349/files#diff-b767582aad69d1a29223fbc6e78492dbba7096d7cc84e39dd7cde1f7797136e0R31 .

I thought that `if config._telemetry_enabled` is `False`, then `telemetry` wouldn't be imported and available and `telemetry.telemetry_writer.add_log()` would give me a `NameError: name 'telemetry' is not defined`, but.. that doesn't appear to be the case.

Reply:
I'm on a roll today. Yes, it is the case. The reason my simple manual test didn't run into that error is precisely because if that `if` statement I added here. 😮‍💨

Reply:
Without the `if`, you get:
```
% DD_INSTRUMENTATION_TELEMETRY_ENABLED=false python -c '\
import ddtrace
with ddtrace.tracer.trace("regression"):
    ddtrace.tracer.configure(partial_flush_min_spans=1)
'
error closing trace
Traceback (most recent call last):
  File "/Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/ddtrace/_trace/span.py", line 627, in __exit__
    self.finish()
  File "/Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/ddtrace/_trace/span.py", line 293, in finish
    self._finish_ns(time_ns())
  File "/Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/ddtrace/_trace/span.py", line 306, in _finish_ns
    cb(self)
  File "/Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/ddtrace/_trace/tracer.py", line 833, in _on_span_finish
    p.on_span_finish(span)
  File "/Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/ddtrace/_trace/processor/__init__.py", line 333, in on_span_finish
    telemetry.telemetry_writer.add_log("WARNING", log_msg)
NameError: name 'telemetry' is not defined
```

Reply:
> Huh. The import of `telemetry` is conditional over at https://github.com/DataDog/dd-trace-py/pull/9349/files#diff-b767582aad69d1a29223fbc6e78492dbba7096d7cc84e39dd7cde1f7797136e0R31 .
> 
> I thought that `if config._telemetry_enabled` is `False`, then `telemetry` wouldn't be imported and available and `telemetry.telemetry_writer.add_log()` would give me a `NameError: name 'telemetry' is not defined`, but.. that doesn't appear to be the case.

oh, I missed that 🤦🏻 

yeah... you are right, the if enabled will need to stay here :/